crm_compassion/models/weekly_demand.py

```diff
@@ -136,6 +154,34 @@ def _default_unsponsored_ambassador(self):
             'crm_compassion.number_children_ambassador'))
         return allocate_per_week - (float(ambass_sponsored) / 52.0)
 
+    @api.multi
+    def _compute_resupply_sub(self):
+        """ Compute SUB resupply. """
+        subs = list()
+        average_sub = self._compute_demand_sub()
+        sub_sponsorships = self.env['recurring.contract'].search([
+            ('origin_id.type', '=', 'sub'),
+        ])
+
+        for sub in sub_sponsorships:
+            if sub.end_date and sub.start_date:
+                start_date = datetime.strptime(sub.start_date, '%Y-%m-%d')
+                end_date = datetime.strptime(sub.end_date, '%Y-%m-%d '
+                                                           '%H:%M:%S')
+                interval = abs((end_date - start_date).days)
+
+                if interval <= 90:
+                    subs.append(sub)
+        rate = len(subs) / average_sub
+
+        cpt = 0
+        for week in self:
+            if cpt < 12 and rate > 0:
+                week.resupply_sub = average_sub * rate
+            else:
+                week.resupply_sub = len(subs)
```

Comment:
That else part is not correct. The result should be the average sub_reject of one year.